acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/runner/Eth2SignerRunner.java

```diff
@@ -0,0 +1,160 @@
+  private final Properties portsProperties;
+
+  private static final String PORTS_FILENAME = "eth2signer.ports";
+  private static final String HTTP_PORT_KEY = "http-port";
+
+  public static Eth2SignerRunner createRunner(final SignerConfiguration signerConfig) {
+    if (Boolean.getBoolean("acctests.runEth2SignerAsProcess")) {
+      return new Eth2SignerProcessRunner(signerConfig);
+    } else {
+      return new Eth2SignerThreadRunner(signerConfig);
+    }
+  }
+
+  public Eth2SignerRunner(final SignerConfiguration signerConfig) {
+    this.signerConfig = signerConfig;
+    this.portsProperties = new Properties();
+
+    if (signerConfig.isDynamicPortAllocation()) {
+      try {
+        this.dataPath = Files.createTempDirectory("acceptance-test");
+      } catch (final IOException e) {
+        throw new RuntimeException(
+            "Failed to create the temporary directory to store the eth2signer.ports file");
+      }
+    } else {
+      dataPath = null;
+    }
+  }
+
+  public void start() {
+    final List<String> params = createCmdLineParams();
+
+    startExecutor(params);
+
+    if (signerConfig.isDynamicPortAllocation()) {
+      loadPortsFile();
+    }
+  }
+
+  protected abstract void startExecutor(final List<String> params);
+
+  public void shutdown() {
+    try {
+      shutdownExecutor();
+    } finally {
+      if (signerConfig.isDynamicPortAllocation()) {
+        try {
+          MoreFiles.deleteRecursively(dataPath, RecursiveDeleteOption.ALLOW_INSECURE);
+        } catch (final IOException e) {
+          LOG.info("Failed to clean up temporary file: {}", dataPath, e);
+        }
+      }
+    }
+  }
+
+  protected abstract void shutdownExecutor();
+
+  public abstract boolean isRunning();
+
+  private List<String> createCmdLineParams() {
+    final String loggingLevel = "DEBUG";
+
+    final List<String> params = new ArrayList<>();
+    params.add("--logging");
+    params.add(loggingLevel);
+    params.add("--http-listen-host");
+    params.add(signerConfig.hostname());
+    params.add("--http-listen-port");
+    params.add(String.valueOf(signerConfig.httpPort()));
+    params.add("--key-store-path");
+    params.add(signerConfig.getKeyStorePath().toString());
+    if (signerConfig.isDynamicPortAllocation()) {
+      params.add("--data-path");
+      params.add(dataPath.toAbsolutePath().toString());
+    }
+
+    return params;
+  }
+
+  private void loadPortsFile() {
+    final File portsFile = new File(dataPath.toFile(), PORTS_FILENAME);
+    LOG.info("Awaiting presence of ethsigner.ports file: {}", portsFile.getAbsolutePath());
+    awaitPortsFile(dataPath);
+    LOG.info("Found ethsigner.ports file: {}", portsFile.getAbsolutePath());
+
+    try (final FileInputStream fis = new FileInputStream(portsFile)) {
+      portsProperties.load(fis);
+      LOG.info("EthSigner ports: {}", portsProperties);
+    } catch (final IOException e) {
+      throw new RuntimeException("Error reading Web3Provider ports file", e);
```

Comment:
nit: There is `UncheckedIOException` in standard JDK which is designed to convert IOexception to runtime exception.